14/umbraco-cms/extending/packages/language-files-for-packages.md

```diff
@@ -2,39 +2,58 @@
 description: "Information on how to use language files to make your Umbraco package UI support multiple languages"
 ---
 
-# Language file for packages
-
-Umbraco Core includes language files, but package authors must provide their own for multi-lingual UI.
-
-## Including new language keys
-
-For each language your package supports, you include an .xml file in the same format as the core language files, named with its language code. The language files must be located in a `Lang` folder inside your package folder in `App_Plugins`. If your package assets are in `/App_Plugins/mypackage` all language files must be placed in the following locations:
-
-- English keys: `/App_Plugins/mypackage/Lang/en-US.xml`
-- Danish keys: `/App_Plugins/mypackage/Lang/da-DK.xml`
-
-{% hint style="info" %}
-The `App_Plugins` version of the `Lang` directory is case sensitive on Linux systems, so make sure that it start with a capital `L`.
-{% endhint %}
+# Languages for packages
+
+If you want your package to be available in different languages, you can use the existing localizations from Umbraco or register your own localizations. The localizations are written as a key-value pair pattern.
+To register localizations to a language, you must add a new manifest to the Extension API. The manifest can be added through the `umbraco-package.json` file like this:
+
+{% code title="umbraco-package.json" lineNumbers="true" %}
+
+```json
+{
+  ...
+  "name": "MyPackage",
+  "extensions": [
+    {
+      "type": "localization",
+      "alias": "MyPackage.Localize.EnUS",
+      "name": "English (United States)",
+      "meta": {
+        "culture": "en-us"
+      },
+      "js": "/App_Plugins/MyPackage/Localization/en-us.js"
+    }
+  ]
+}
+```
 
 ## Language file format
 
-Each language file can include one or more area. Each area contains a collection of language keys with the translation.
-
-For reference on the language file format see the core [language files on GitHub](https://github.com/umbraco/Umbraco-CMS/tree/v9/dev/src/Umbraco.Web.UI/umbraco/config/lang)
+## Layout of the Localization Files
 
-### Sample structure
-
-```xml
-<?xml version="1.0" encoding="utf-8" standalone="yes"?>
-<language alias="da" intName="Danish" localName="dansk" lcid="6" culture="da-DK">
-    <creator>
-        <name>The Umbraco community</name>
-        <link>http://mydomain.com</link>
-    </creator>
-    <area alias="dialog">
-        <key alias="myKey">Min nøgle</key>
-        <key alias="otherKey">Min anden nøgle</key>
-    </area>
-</language>
+The localization files for the UI are JS modules with a default export containing a key-value structure organized in sections.
+```Javascript
+export default {
+ section: {
+  key1: 'value1',
+  key2: 'value2',
+ },
+};
+```
+The sections and keys will be formatted into a map in Umbraco with the format section_key1 and section_key2. These form the unique key they are requested.
```

Comment:
Tried to rewrite this. Not sure it's correct though?

```suggestion

The sections and keys will be formatted into a map in Umbraco with the format `section_key1` and `section_key2`. These form the unique key that it requested.
```

Reply:
Looks fine to me :D